data/src/new_etl/classes/featurelayer.py

```diff
@@ -118,75 +117,152 @@ def check_psql(self):
 
     def load_data(self):
         log.info(f"Loading data for {self.name} from {self.type}...")
+
         if self.type == "gdf":
-            pass
-        else:
-            try:
-                if self.type == "esri":
-                    if not self.esri_rest_urls:
-                        raise ValueError("Must provide a URL to load data from Esri")
-
-                    gdfs = []
-                    for url in self.esri_rest_urls:
-                        print(f"Processing URL: {url}")  # Debugging: Print the URL
-
-                        # Use EsriDumper to get features
-                        dumper = EsriDumper(url)
-                        features = [feature for feature in dumper]
-
-                        if not features:
-                            log.error(f"No features returned for URL: {url}")
-                            continue
-
-                        geojson_features = {
-                            "type": "FeatureCollection",
-                            "features": features,
-                        }
-                        gdf = gpd.GeoDataFrame.from_features(
-                            geojson_features, crs=self.input_crs
-                        )
-                        gdf = gdf.to_crs(self.crs)
+            return  # Skip processing for gdf type
+
+        try:
+            if self.type == "esri":
```

Comment:
This is nested several levels deep. It might make sense to have some helper functions. `process_esri_data`  and `process_carto_data` seem like logical choices.